salt/metalk8s/kubeadm/init/preflight/mandatory.sls

```diff
@@ -0,0 +1,70 @@
+{%- from "metalk8s/map.jinja" import defaults with context %}
+{%- from "metalk8s/map.jinja" import kubeadm_preflight with context %}
+{%- from "metalk8s/map.jinja" import kubelet with context %}
+
+Install mandatory packages:
+  pkg.installed:
+    - pkgs: {{ kubeadm_preflight.mandatory.packages }}
+
+{%- if kubelet.container_engine %}
+Enable {{ kubelet.container_engine }} service:
+  service.running:
+    - name: {{ kubelet.container_engine }}
+    - enable: true
+    - require_in:
+      - test: Check that the crictl socket answer
+{%- endif %}
+
+# required to set sysctl net.bridge.bridge-nf-call-iptables
+Add the module br_netfilter to kernel:
+  kmod.present:
+    - name: br_netfilter
+    - persist: True
+
+{%- for item, value in kubeadm_preflight.mandatory.sysctl_values.items() %}
+Set sysctl {{ item }} value to {{ value }}:
+  sysctl.present:
+    - name: {{ item }}
+    - value: {{ value }}
+    - config: /etc/sysctl.d/60-metalk8s.conf
+    {%- if item in ("net.bridge.bridge-nf-call-ip6tables", "net.bridge.bridge-nf-call-iptables") %}
+    - require:
+      - kmod: Add the module br_netfilter to kernel
+    {%- endif %}
+{%- endfor %}
+
+{%- for swap_device in salt.mount.swaps().keys() %}
+Disable swap device {{ swap_device }}:
+  module.run:
+    - name: mount.swapoff
+    - m_name: {{ swap_device }}
+{%- endfor %}
+
+Prevent swap mount from fstab:
+  file.comment:
+    - name: /etc/fstab
+    - regex: .+\s+.+\s+swap\s+.+
+    - onlyif:
+      # Add a condition to avoid the state failure
+      # in case we don t have swap in /etc/fstab
+      - test -n "$(awk '$3=="swap" {print}' /etc/fstab)"
+
+Check that the crictl socket answer:
+  # use test.succeed_without_changes instead of cmd.run for idempotency
+  test.succeed_without_changes:
+    - name: crictl info
+    - check_cmd:
+      - crictl info
```

Comment:
Nit, and maybe should be tracked as a tech-debt rule: whilst now we have `/etc/crictl.yaml`, in this specific case we *must* check whether the CRI daemon is listening on the path we believe it should be (i.e. the path we'll pass to `kubelet`). We can't rely on a value in `/etc/crictl.yaml`.

So in here, we *must* pass `--runtime-endpoint` to the `crictl` invocation.